pkg/controller/job/job_controller.go

```diff
@@ -1604,7 +1607,9 @@ func (jm *Controller) manageJob(ctx context.Context, job *batch.Job, jobCtx *syn
 						jm.expectations.CreationObserved(logger, jobKey)
 						atomic.AddInt32(&active, -1)
 						errCh <- err
+						creationsFailed++
```

Comment:
you need to use `atomic.AddInt32`, because this is happening in a routine.

Reply:
good catch